packages/block-library/src/cover/edit.native.js

```diff
@@ -145,6 +168,10 @@ const Cover = ( {
 			requestImageUploadCancelDialog( id );
 		} else if ( shouldShowFailure ) {
 			requestImageFailedRetryDialog( id );
+		} else if ( backgroundType === MEDIA_TYPE_IMAGE && isMediaSelected ) {
+			requestImageFullscreenPreview( url );
+		} else if ( backgroundType === MEDIA_TYPE_IMAGE ) {
+			setMediaSelected( true );
```

Comment:
Minor nit: I wonder if it'd be more readable to only test the `backgroundType` once, e.g.:

```suggestion
		} else if ( backgroundType === MEDIA_TYPE_IMAGE ) {
			if ( isMediaSelected ) {
				requestImageFullscreenPreview( url );
			} else {
				setMediaSelected( true );
			}
```

Reply:
Thanks for the suggestion Matt! I'll wait for the design feedback to see if this logic needs to change but if it doesn't I'll use that =)